cmd/kube-scheduler/app/options/options.go

```diff
@@ -247,13 +248,20 @@ func (o *Options) Config() (*schedulerappconfig.Config, error) {
 		return nil, err
 	}
 
-	coreBroadcaster := record.NewBroadcaster()
+	c.EventClient = events.NewEventClient(eventClient)
+
+	// Set CoreClient and CoreBroadcaster for leader election if they are nil
+	// since leader election hasn't been migrated to use the new Event API.
+	if c.EventClient.CoreBroadcaster == nil {
```

Comment:
add a TODO to get rid of that once leader election is migrated.

Reply:
Actually I take it back. I don't like the fact that CoreBroadcaster is a public field.
I will comment on the original PR.

Reply:
I have a feeling that this won't even compile - EventClient is an interface that doesn't have CoreBroadcaster field...

Reply:
Updated the codes.